### PR TITLE
Serve files without using FileStreamer-like object

### PR DESCRIPTION
**Pull request by lfidnl**
Hello!

I've implemented serving file without using FileStremer-like object. Now user can use only path to file and Grape send file directly or will use `Rack::Sendfile` middleware if it turned on.

I've saved backward compatibility with FileStreamer-like object with displaying deprecation warning.

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -42,6 +42,8 @@ Metrics/MethodLength:
 # Configuration parameters: CountComments.
 Metrics/ModuleLength:
   Max: 272
+  Exclude:
```
Comment (dblock) [inline .rubocop_todo.yml]: You might want to just run `rubocop --auto-gen-config` for this.

File: lib/grape/serve_file/file_body.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      # Need for Rack::Sendfile middleware
+      #
+      # @return [String]
+      def to_path
+        path
+      end
+
+      def each
+        File.open(path, 'rb') do |file|
+          while (chunk = file.read(16_384))
```
Comment (dblock) [inline lib/grape/serve_file/file_body.rb]: Should probably be extracted as a constant, just in case someone wants to update it at their runtime.

**Comment by dblock** [comment]
Looks great. Needs a CHANGELOG entry please.

**Comment by lfidnl** [comment]
Thanks for response! Ohhh sorry, forgot about CHANGELOG

File: lib/grape/serve_file/file_body.rb (hunk 2 above)
Comment (lfidnl) [inline lib/grape/serve_file/file_body.rb]: As I understood, you mean create constant for 16384 number, is it?

Comment (dblock) [inline lib/grape/serve_file/file_body.rb]: Yep

**Comment by lfidnl** [comment]
Updated according to your comments

File: lib/grape/dsl/inside_route.rb [hunk 3]
```diff
@@ -191,8 +191,12 @@ def body(value = nil)
       #
       #   GET /file # => "contents of file"
       def file(value = nil)
-        if value
-          @file = Grape::Util::FileResponse.new(value)
+        if value.is_a?(String)
+          file_body = Grape::ServeFile::FileBody.new(value)
+          @file = Grape::ServeFile::FileResponse.new(file_body)
+        elsif !value.is_a?(NilClass)
+          warn '[DEPRECATION] Argument as FileStreamer-like object will be deprecated. Use path to file instead'
```
Comment (dblock) [inline lib/grape/dsl/inside_route.rb]: In other places we say "is deprecated", not "will be" and usually make this a full sentence with a period in the end. Like this: https://github.com/ruby-grape/grape/blob/4032ea8f7747483b84663c260d4aba0491bd7777/lib/grape/dsl/desc.rb#L50.

**Comment by dblock** [comment]
One last thing this needs is an entry in UPGRADING.md, please.

File: lib/grape/dsl/inside_route.rb (hunk 3 above)
Comment (lfidnl) [inline lib/grape/dsl/inside_route.rb]: I mean that support of argument as FileSteam-like object will be removed from Grape in next version after 0.15.1 or 0.16.0. After releasing 0.15.1 or 0.16.0 I would like to remove it at all. Sorry for my English :)
OK, I'll replace it with:

```
[DEPRECATION] Argument as FileStreamer-like object is deprecated. Use path to file instead.
```

Comment (dblock) [inline lib/grape/dsl/inside_route.rb]: Thanks. If you're rubbing the bext version the old code is already deprecated but it does work with a warning. I think present tense is correct :) thanks for making the changes!

**Comment by dblock** [comment]
Last thought, don't feel that strongly about it, but maybe ServeFile should just be File?

**Comment by lfidnl** [comment]
I chose `ServeFile` to avoid confusion with the regular ruby `File` class. I know that we have namespaces and etc, but I think it will be better :)

**Comment by dblock** [comment]
:+1:

File: UPGRADING.md [hunk 4]
```diff
@@ -23,6 +23,12 @@ TwitterAPI::routes[0].settings[:custom] # => { key: 'value' }
 TwitterAPI::routes[0].request_method # => 'GET'
 ```
 
+#### `file` method accepts path to file
+
+Now to serve files via Grape just pass the path to the file. Functionality with FileStreamer-like objects is deprecated.
+
+Please, replace your FileStreamer-like objects with paths of served files.
```
Comment (dblock) [inline UPGRADING.md]: Care to provide an example here? Thanks.

Comment (lfidnl) [inline UPGRADING.md]: No problem :)

**Comment by dblock** [comment]
Merged, thanks.

**Pull request merged**